announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -1716,6 +1828,12 @@ else if (channel.getGroupsAllowAddMessage().size() > 0)
 							// to group otherwise
 							context.put("announceTo", "groups");
 						}
+					} else {
+						state.setTempAnnounceToRoles(
+							Arrays.asList(
+								state.getEdit().getProperties().getProperty("selectedRoles").replaceAll("[\\[\\]]", "").replaceAll(", ", ",").split(",")
```

Comment:
Use the constant and explain why you do the replaceAll.

Reply:
I have used that for a logic that I am going to change so that I do not have to do it all the time